File: VBA/sample02.vb
REM https://www.fe-siken.com/kakomon/sample/b2.html

Public Function fizzBuzz(num)
    If num Mod 15 = 0 Then
        fizzBuzz = "3と5で割り切れる"
    ElseIf num Mod 3 = 0 Then
        fizzBuzz = "3で割り切れる"
    ElseIf num Mod 5 = 0 Then
        fizzBuzz = "5で割り切れる"
    Else
        fizzBuzz = "3でも5でも割り切れない"
    End If
End Function

Public Sub myPrint()
    For i = 1 To 15
        Debug.Print ("i = " & i & " " & fizzBuzz(i))
    Next
End Sub

File: Java/Sample02.java
// https://www.fe-siken.com/kakomon/sample/b2.html

class Sample02 {
  public static void main(String[] args) {
    for (int i = 1; i < 16; i++) {
      String a = "i = " + String.valueOf(i) + " " + fizzBuzz(i);
      System.out.println(a);
    }
  }

  public static String fizzBuzz(int i) {
    if (i % 15 == 0) {
      return "3と5で割り切れる";
    } else if (i % 3 == 0) {
      return "3で割り切れる";
    } else if (i % 5 == 0) {
      return "5で割り切れる";
    } else {
      return "3でも5でも割り切れない";
    }
  }
}
